lib/prompts/selectAppPrompt.ts

```diff
@@ -0,0 +1,53 @@
+import { logger } from '@hubspot/local-dev-lib/logger';
+import { fetchPublicAppsForPortal } from '@hubspot/local-dev-lib/api/appsDev';
+import { PublicApp } from '@hubspot/local-dev-lib/types/Apps';
+import { debugError } from '../errorHandlers/index';
+import { lib } from '../../lang/en';
+import { EXIT_CODES } from '../enums/exitCodes';
+import { listPrompt } from '../prompts/promptUtils';
+
+export async function selectAppPrompt(
+  accountId: number,
+  appId?: number
+): Promise<PublicApp> {
+  let availableApps: PublicApp[] = [];
+
+  try {
+    const appsResponse = await fetchPublicAppsForPortal(accountId);
+    if (appsResponse.data.results) {
+      availableApps = appsResponse.data.results;
+    }
+  } catch (err) {
+    debugError(err);
+  }
+
+  if (availableApps.length === 0) {
+    logger.error(lib.prompts.selectAppPrompt.errors.noApps);
+    process.exit(EXIT_CODES.ERROR);
```

Comment:
Just food for thought.  I've been trying to avoid calling `process.exit` outside of the handler methods and throwing errors instead.  The handler can then catch/log/exit.  I feels like that should be the responsibility of the handler and not the prompt method.  It also makes these methods easier to test.

Reply:
Yeah that's a good call. I updated this to just return null instead, and then exit from the handler if there's no app returned in [ad1136d](https://github.com/HubSpot/hubspot-cli/pull/1537/commits/ad1136dbf2636380611e6db491b3ed6653c7e7f9)